core/controllers/base.py

```diff
@@ -166,33 +166,32 @@ def __init__(self, request, response):  # pylint: disable=super-init-not-called
             self.payload = None
         self.iframed = False
 
-        self.is_super_admin = user_services.is_current_user_super_admin()
+        auth_claims = user_services.get_auth_claims_from_request(request)
+        self.is_super_admin = auth_claims is not None and auth_claims.is_admin
+
         if feconf.ENABLE_MAINTENANCE_MODE and not self.is_super_admin:
             return
 
-        self.gae_id = user_services.get_current_gae_id()
         self.user_id = None
         self.username = None
         self.partially_logged_in = False
         self.user_is_scheduled_for_deletion = False
 
-        # TODO(#11462): This part should be moved to the service layer when we
-        # migrate to Firebase.
```

Comment:
I think we expected to move the whole code after `if` (and before line 196)

Reply:
Sorry, I'm not sure I follow. Why do we want to move everything? I thought we just wanted to get rid of the GAE-specific code?

Reply:
Oh, I didn't realize that the stuff is now platform-agnostic, it should be fine then.